src/application/backend.ts

```diff
@@ -382,19 +385,57 @@ export default class Backend {
                 return handleError(id, e);
               }
 
+            case Marina.prototype.getTransactions.name:
+              try {
+                if (!(await this.isCurentSiteEnabled())) {
+                  return handleError(id, new Error('User must authorize the current website'));
+                }
+                const transactions = walletTransactions(marinaStore.getState());
+                return handleResponse(id, transactions);
+              } catch (e) {
+                return handleError(id, e);
+              }
+
+            case Marina.prototype.getCoins.name:
+              try {
+                if (!(await this.isCurentSiteEnabled())) {
+                  return handleError(id, new Error('User must authorize the current website'));
+                }
+                const coins = Object.values(marinaStore.getState().wallet.utxoMap);
```

Comment:
Here we get the whole state, after we use selector. Would be better to access state same way